platform/wayland/cog-platform-wl.c

```diff
@@ -568,44 +565,64 @@ pointer_on_enter(void              *data,
                  wl_fixed_t         fixed_x,
                  wl_fixed_t         fixed_y)
 {
-    CogWlPlatform *platform = data;
-    CogWlDisplay  *display = platform->display;
+    CogWlSeat *seat = data;
 
-    display->seat.keyboard.serial = serial;
-    display->seat.pointer.serial = serial;
-    display->seat.pointer.surface = surface;
+    if (pointer != seat->pointer_obj) {
+        g_critical("%s: Got pointer %p, expected %p.", G_STRFUNC, pointer, seat->pointer_obj);
+        return;
+    }
```

Comment:
These are additional safe checks that they have been added in the input events handlers just to be sure the real implementation meets with the expectations.